packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -610,15 +610,43 @@ export class NetworkMonitor {
         )
       }
 
-      // Check for validity with epoch manager currentEpoch, and fetch block hash
-      const validBlock =
-        result.data.network.latestValidBlockNumber.epochNumber ==
-        (await this.currentEpochNumber())
-          ? result.data.network.latestValidBlockNumber
-          : result.data.network.latestValidBlockNumber.previousBlockNumber
-
-      // Resolve block hash for the given block number.
-      // Calls the configured provider for blocks from protocol chain, or Graph Node otherwise.
+      // Check for validity with Epoch Manager's currentEpoch method, and fetch block hash after that.
+      const epochManagerCurrentEpoch = await this.currentEpochNumber()
+      const epochSubgraphCurrentEpoch =
+        result.data.network.latestValidBlockNumber.epochNumber
+
+      // The Epoch Manager will always lead new epochs, so it should be minuend.
+      const epochDifference = epochManagerCurrentEpoch - epochSubgraphCurrentEpoch
+
+      if (epochDifference < 0) {
+        // Somehow, the Epoch Manager is behind the Epoch Subgraph. This is a critical failure.
+        const criticalErrorMessage = `The Epoch Manager is behind the Epoch Subgraph`
+        this.logger.critical(criticalErrorMessage, {
+          epochManagerCurrentEpoch,
+          epochSubgraphCurrentEpoch,
+        })
+        throw new Error(criticalErrorMessage)
+      }
+
+      if (epochDifference > 1) {
+        // It is not acceptable for the Epoch Subgraph to be more than one epoch away from the
+        // Epoch Manager.
+        const errorMessage = `The Epoch Subgraph is ${epochDifference} epochs away from the Epoch Manager`
+        this.logger.error(errorMessage)
+        throw new Error(errorMessage)
+      }
+
+      if (epochDifference === 1) {
+        // It is acceptable that the Epoch Subgraph stays at least one epoch behind the Epoch
+        // Manager, considering this measurement could have happened during an epoch transition.
+        this.logger.info(
+          `Considering the penultimate epoch as the Epoch Subgraph is one epoch behind the Epoch Manager`,
+        )
+      }
+
+      // At this point, the epoch difference is either 0 or 1.
+      const validBlock = result.data.network.latestValidBlockNumber
+
       let startBlockHash: string
       if (networkID == this.networkCAIPID) {
         startBlockHash = (await this.ethereum.getBlock(+validBlock.blockNumber)).hash
```

Comment:
Maybe we can add a log at the end of the function with the results?

```
this.logger.info('Resolved current Epoch', {
    network: networkAlias,
    currentEpoch: validBlock.epochNumber,
    currentEpochstartBlockNumber: +validBlock.blockNumber,
    epochDifferenceFromContracts: epochDifference,
    latestBlock: latestBlock,

Reply:
Fixed in 299e359c